lua/lean/infoview.lua

```diff
@@ -154,7 +154,34 @@ function Info:widget()
   raw_pos = raw_pos + pos[2] + 1
 
   local _, div_stack = self.div:div_from_pos(raw_pos)
-  require"vim.lsp.util".open_floating_preview(vim.split(vim.inspect(div_stack[#div_stack]), "\n"), nil, {})
+
+  local sess = nil
+  for _, div in ipairs(div_stack) do
+    sess = sess or div.tags.rpc_sess
+  end
+
+  for i = #div_stack, 1, -1 do
+    local info_with_ctx = div_stack[i].tags.info_with_ctx
+    if info_with_ctx ~= nil then
+      return a.void(function()
+        local info_popup = sess:infoToInteractive(info_with_ctx)
```

Comment:
CMIIW but I think the div is the only thing with the reference to this, so it could get GC'ed immediately after making the popup. I think you may have not noticed this since it looks like these popups aren't interactive yet (will soon try this myself).